app/Http/Controllers/Teams/MembersController.php

```diff
@@ -48,4 +48,24 @@ public function index(string $teamId): Response
 
         return ext_view('teams.members.index', compact('team'));
     }
+
+    public function setLeader(string $teamId, string $userId): Response
+    {
+        $teamMember = TeamMember::where([
+            'team_id' => $teamId,
+            'user_id' => $userId,
+        ])->firstOrFail();
+        $team = $teamMember->team;
+
+        priv_check('TeamUpdate', $team)->ensureCan();
+
+        $team->update(['leader_id' => $teamMember->user_id]);
+
+        \Session::flash('popup', osu_trans(
+            'teams.members.set_leader.success',
+            ['user' => $teamMember->userOrDeleted()->username],
```

Comment:
does it even make sense to transfer to a deleted or restricted user?

Reply:
I'm not quite sure where to put the validation/authorization (which one is it even) so it's inlined in the controller action for now ಠ_ಠ